src/index.ts

```diff
@@ -0,0 +1,165 @@
+import * as crypto from 'crypto'
+import * as path from 'path'
+import { EventEmitter } from 'events'
+const createLogger = require('ilp-logger')
+const log = createLogger('ilp-plugin')
+require('source-map-support').install()
+
+interface FunctionWithVersion extends Function {
+  version?: number
+}
+export interface ConnectOptions {
+  timeout?: number
+}
+export type DataHandler = (data: Buffer) => Promise<Buffer>
+export type MoneyHandler = (amount: string) => Promise<void>
+export interface PluginInstance extends EventEmitter {
+  constructor: FunctionWithVersion
+  connect: (options?: ConnectOptions) => Promise<void>
+  disconnect: () => Promise<void>
+  isConnected: () => boolean
+  sendData: DataHandler
+  sendMoney: MoneyHandler
+  registerDataHandler: (handler: DataHandler) => void
+  deregisterDataHandler: () => void
+  registerMoneyHandler: (handler: MoneyHandler) => void
+  deregisterMoneyHandler: () => void
+  getAdminInfo? (): Promise<object>
+  sendAdminInfo? (info: object): Promise<object>
+}
+
+/**
+ * A type guard for ILP plugins
+ *
+ * See https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards
+ *
+ * @param plugin a class to test
+ */
+export function isPlugin (plugin: any): plugin is PluginInstance {
+  return plugin.connect !== undefined &&
+    plugin.disconnect !== undefined &&
+    plugin.isConnected !== undefined &&
+    plugin.sendData !== undefined &&
+    plugin.sendMoney !== undefined &&
+    plugin.registerDataHandler !== undefined &&
+    plugin.deregisterDataHandler !== undefined &&
+    plugin.registerMoneyHandler !== undefined &&
+    plugin.deregisterMoneyHandler !== undefined
+}
+
+export const DEFAULT_PLUGIN_MODULE = 'ilp-plugin-btp'
+
+export interface PluginServices {
+  store: any
+  log: Logger
```

Comment:
`Logger` needs to be imported from `ilp-logger`.
Could use the latest version of `ilp-logger` (which has Typescript) while you're at it.